### PR TITLE
Bats readme

### PR DESCRIPTION
**Pull request by IsaSih**
Created a readme providing documentation on the how to: setup, run, write and lint BATS tests

### CODE REVIEW

File: bats/README.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+          ```
+          go install mvdan.cc/sh/v3/cmd/shfmt@v3.6.0
+          ```
+
+  2. Get the syntax and formatting feedback for BATS linting by running:
+
+      ```
+      make -C bats lint
+      ```
+3. Please, make sure to fix the highlighted linting errors prior to submitting a PR. Not required, but highly recommendable. To do so, run the following command:
```
Comment (jandubois) [inline bats/README.md:123]: Fixing linter errors is **required** now; Github actions will fail if any errors are detected.

```suggestion
3. Please, make sure to fix the highlighted linting errors prior to submitting a PR. You can automatically apply formatting changes suggested by `shfmt` by running the following command:
```

**Review by jandubois — CHANGES_REQUESTED** [review]
Thanks! I do have some change requests... 😄

**Comment by jandubois** [comment]
Please also add a link from `docs/development/README.md` to `bats/README.md` (similar to the link for the screenshots README).

File: bats/README.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+      ```
+      cd bats
+      RD_CONTAINER_RUNTIME=moby RD_USE_IMAGE_ALLOW_LIST=false ./bats-core/bin/bats tests/registry/creds.bats
+      ```
+
+## Writing BATS Tests
+
+1. Add BATS test by creating files with `.bats` extension under ./bats/tests/FOLDER_NAME
+2. A Bats test file is a Bash script with special syntax for defining test cases. Under the hood, each test case is just a function with a description.
```
[the remark this answers is not in the document]
Comment (IsaSih) [reply bats/README.md:45]: do you want me to remove it or to keep this as a simple brief intro the way it is?

[the remark this answers is not in the document]
Comment (jandubois) [reply bats/README.md:48]: I would remove it unless you have some good argument why you want to keep it. It is not self-explanatory, so you still need to read the docs anyways. And we have plenty of tests that could work as an example.

Maybe have one example extensively documented, and then link to the actual test from here as a "recommended tutorial"? But maybe this should wait until later. Just get a README published, and then refine over time.

File: bats/README.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
+## Overview
+
+BATS is a testing framework for Bash shell scripts that provides supporting libraries and helpers for customizable server-side test automation.
```
Comment (jandubois) [inline bats/README.md:3]: Not sure what "server-side" means here; Rancher Desktop is a desktop application.

```suggestion
BATS is a testing framework for Bash shell scripts that provides supporting libraries and helpers for customizable test automation.
```

File: bats/README.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+## Setup
+
+It's important to have a Rancher Desktop CI build installed with privileged rights before running the BATS tests.
```
Comment (jandubois) [inline bats/README.md:7]: Let's remove "with privileged rights" here, assuming it just means "installed for all users". I'll create a PR to fall back on a per-user installation if the system-wide location does not exist.

```suggestion
It's important to have a Rancher Desktop CI or release build installed before running the BATS tests.
```

File: bats/README.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+On Windows:
+  Prior to running `git submodule update --init`, it's important to setup the Git configuration by running the following commands:
+
```
Comment (jandubois) [inline bats/README.md:11]: I think you should explain that you need to clone the repo first.

Note that changing crlf settings is not necessary when you clone inside a WSL distro; it is only needed when you clone on Win32.

File: bats/README.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+All platforms:
+From the root directory of the Git repository, run the following commands to install BATS and its helper libraries into the BATS test directory:
+
+  ```
+  git submodule update --init
+  ```
+
+## Running BATS
+
+1. To run the BATS test, specify the path to BATS executable from bats-core and run the following commands:
```
Comment (jandubois) [inline bats/README.md:27]: Sine you talk above about cloning the repo on Windows, you should also explain how to locate it inside a WSL distro on `/mnt/...`.

I see that happens in the 3rd example, but it does not explain that `/mnt/c` represents the `C:` drive from windows.

**Review by jandubois — CHANGES_REQUESTED** [review]

File: bats/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (IsaSih) [reply bats/README.md:60]: okay, I deleted the existing example and kept only the links to the documentation. I can elaborate a tutorial in a follow up PR

**Review by gunamata — APPROVED** [review]
LGTM

**Review by jandubois — APPROVED** [review]

**Pull request merged**